src/parsers/manifestjson.js

```diff
@@ -797,6 +801,13 @@ export default class ManifestJSONParser extends JSONParser {
     }
   }
 
+  validateHomePageURL(url) {
+    const restrictedDomain = 'addons-dev.allizom.org';
```

Comment:
Lets create an array or set of urls exported by `src/const.js`, `addons-dev.allizom.org` is the url reported by QA because they verify the fixes on the dev server, but we should also (and mainly) reject homepage_url set to 'addons.mozilla.org'.